colossalai/inference/engine/infer_struct.py

```diff
@@ -0,0 +1,141 @@
+import enum
+from dataclasses import dataclass
+from typing import Dict, List, Set
+
+
+class RequsetStatus(enum.Enum):
+    """The status of Sentences"""
+
+    WAITING = enum.auto()
+    RUNNING = enum.auto()
+    ABORTED = enum.auto()
+    OVERLENGTH = enum.auto()
+    COMPLETED = enum.auto()
+    LENGTH_CAPPED = enum.auto()
+
+    @staticmethod
+    def is_finished(status: "SentenceStatus") -> bool:
+        return status in [
+            OVERLENGTH,
+            COMPLETED,
+            LENGTH_CAPPED,
+        ]
+
+    @staticmethod
+    def is_running(status: "SentenceStatus") -> bool:
+        return status == RUNNING
+
+    @staticmethod
+    def is_waiting(status: "SentenceStatus") -> bool:
+        return status == WAITING
+
+
+class Sequence:
+    """Store the information of a input Sequence.
+
+    Args:
+        request_id: The ID of the sequence.
+        prompt: The prompt of the sequence.
+        token_id: The ID of the sequence.
+        block_size: The block size of the sequence.
+        sample_params: The sample_params of the sequence.
+        block_table_index: The index of this sequence in block_table.
+    """
+
+    def __init__(
+        self,
+        request_id: int,
+        prompt: str,
+        token_id: int,
+        blokc_size: int,
+        sample_params: SampleParams,
+        block_table_index: int,
+    ):
+        self.request_id = request_id
+        self.input_token_id = token_id
+        self.prompt = prompt
+        self.blokc_size = blokc_size
+        self.output_token_id = []
+        self.output = ""
+        self.status = SentenceStatus.WAITING
+        self.sample_params = sample_params
+        self.batch_infer_state = batch_infer_state
+        self.block_table_index = block_table_index
+
+    def get_sentence_len(self) -> None:
+        return len(self.input_token_id) + len(self.output_token_id)
+
+    def get_input_len(self) -> None:
+        return len(self.input_token_id)
+
+    def get_output_len(self) -> None:
+        return len(self.output_token_id)
+
+    def check_finish(self) -> bool:
+        return SentenceStatus.check_finish(self.status)
+
+    def __repr__(self) -> str:
+        return (
+            f"Request ID(request_id={self.request_id}, "
+            f"prompt={self.prompt}, "
+            f"status={self.status.name}, "
+            f"sample_params={self.sample_params}, "
+            f"logical block number={len(self._logical_blocks)}"
+        )
+
+
+@dataclass
+class BatchInferState:
```

Comment:
our batch is dynamic now , may be change a name ?

Reply:
Done